### PR TITLE
Fix tests to run on Cairo 1.18.0

### PR DESCRIPTION
**Pull request by naveen521kk**
## Overview: What does this pull request change?

Fixes https://github.com/ManimCommunity/manim/issues/3310. I've regenerated all the tests so that it passes with Cairo 1.18.0 and also updated CI so that that version is used.

## Further Information and Comments

~I still have to figure out how to skip tests when the cairo version is <1.18.0.~
Done. The tests will be skipped when cairo <1.18.0 is found.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: scripts/ci_build_cairo.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
+# Logic is as follows:
```
Comment (behackl) [inline scripts/ci_build_cairo.py:1]: Should this file perhaps live in, say, `.github/scripts`? Practically, it's only for CI and not something one should use locally, I guess. 🤔

File: manim/utils/testing/frames_comparison.py [hunk 2]
```diff
@@ -81,6 +83,9 @@ def decorator_maker(tested_scene_construct):
         @functools.wraps(tested_scene_construct)
         # The "request" parameter is meant to be used as a fixture by pytest. See below.
         def wrapper(*args, request: FixtureRequest, tmp_path, **kwargs):
+            # check for cairo version
+            if renderer_class is CairoRenderer and cairo.cairo_version() < 11800:
```
Comment (behackl) [inline manim/utils/testing/frames_comparison.py:87]: The discrepancy between 11800 here and the `1.18.0` string in the build script is somewhat unfortunate -- but should this here perhaps be stored in a (constant) variable too?

**Review by behackl — APPROVED** [review]
This looks fine, and the fact that the pipeline passes speaks for itself. I've left two minor comments, please take a look -- otherwise I'm all for merging this. Thanks for your efforts!

File: scripts/ci_build_cairo.py (hunk 1 above)
Comment (naveen521kk) [reply scripts/ci_build_cairo.py:1]: Yeah, thinking of it, I think I'll move this to the other folder.

**Comment by behackl** [comment]
Thank you! 🚀

**Comment by cameronrutherford** [comment]
For context I am trying to install and do some cool work with revealJS in a dev-container, and ended up here. Three questions:

- If cairo was updated in tests, why not also update the pyproject.toml?
- Should there be documentation about installing with poetry? I naively tried to do so and had to find the secret poetry flag that fixes this https://github.com/ManimCommunity/manim/blob/main/.github/workflows/ci.yml#L166
- Why does the cairo version need to be pinned?

**Comment by cameronrutherford** [comment]
> * If cairo was updated in tests, why not also update the pyproject.toml?

Answering this myself, pycairo != cairo, and so pycairo looks for a cairo install at runtime.

> * Should there be documentation about installing with poetry? I naively tried to do so and had to find the secret poetry flag that fixes this https://github.com/ManimCommunity/manim/blob/main/.github/workflows/ci.yml#L166

I guess this is likely because of the extra cairo dep that isn't captured. This is captured in conda package and other distributions, but sticks out installing with poetry for some reason. Maybe I should just try pip?

Is there any interest in a spack.io package for manim? Since there is an existing cairo https://packages.spack.io/package.html?name=cairo and pycairo https://packages.spack.io/package.html?name=py-pycairo package, would be relatively easy I would hope.

> * Why does the cairo version need to be pinned?

Still unclear on this - what are the changes between cairo versions that are significant?

**Comment by naveen521kk** [comment]
> Once fixing this I had errors where cairo was missing. I guess this is likely because of the extra cairo dep that isn't captured. This is captured in conda package and other distributions, but sticks out installing with poetry for some reason. Maybe I should just try pip?

There is a conda package for manim, https://anaconda.org/conda-forge/manim. You would only need to use poetry only when you're developing manim itself and not otherwise.

> Is there any interest in a spack.io package for manim? Since there is an existing cairo https://packages.spack.io/package.html?name=cairo and pycairo https://packages.spack.io/package.html?name=py-pycairo package, would be relatively easy I would hope.

Never heard of spack.io before and not sure how it'll help people. 

>  Why does the cairo version need to be pinned?

There was a new subpixel antialiasing introduced in newer versions of cairo which started to fail the test suite. That's why it needed an update as in this PR. From a user perspective, there is no change and the cairo version doesn't need to be pinned.

**Pull request merged**